cleverhans/utils_th.py

```diff
@@ -62,6 +62,39 @@ def adadelta(loss_or_grads, params, learning_rate=1.0, rho=0.95, epsilon=1e-6):
 
     return updates
 
+def adam(loss_or_grads, params, learning_rate=0.001, beta1=0.9,
```

Comment:
Your `carlini_l2_step` already imports Keras though, so you do have a Keras dependency either way.